src/bignum.nr

```diff
@@ -272,11 +270,17 @@ where
         let params = Params::get_params();
         evaluate_quadratic_expression::<_, MOD_BITS, _, _, _, _>(
             params,
-            map(lhs_terms, |bns| map(bns, |bn| Self::get_limbs(bn))),
+            map(
```

Comment:
@TomAFrench I've honed in on this PR being the one that caused a regression in the constraint counts of BigNum. In particular, I was measuring the constraints of calls to this `evaluate_quadratic_expression` function. 
Any thoughts on why this is increasing constraint counts? Is it simply that converting between slices and arrays is not free?

Reply:
The extra ACIR opcodes (for a `main` function which calls this `evaluate_quadratic_expression`) function appear to be:
```
INIT (id: 0, len: 4) 
EXPR [ (-1, _5) 0 ]
MEM (id: 0, read at: x5, value: x6) 
EXPR [ (-1, _7) 1 ]
MEM (id: 0, read at: x7, value: x8) 
EXPR [ (-1, _9) 2 ]
MEM (id: 0, read at: x9, value: x10) 
EXPR [ (-1, _11) 3 ]
MEM (id: 0, read at: x11, value: x12) 
INIT (id: 1, len: 4) 
MEM (id: 1, read at: x5, value: x13) 
MEM (id: 1, read at: x7, value: x14) 
MEM (id: 1, read at: x9, value: x15) 
MEM (id: 1, read at: x11, value: x16) 
```

Reply:
Yeah, this is bad actor codegen as we're just reading all the values out of the array to write into another one.

Reply:
Neater code here, to fix it: https://github.com/noir-lang/noir-bignum/pull/53